### PR TITLE
DYN-6440 Custom Selection Overlapped

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

Fixing problem with large text in the Custom Selection nodes.
When choosing a large string text from the ComboBox was overlapping other controls at left side then for fixing this bug I've added a new TextBlock that will hidden for all the nodes but will be shown just for Custom Selection nodes.
This new TextBlock will show "..." at the end of the string when trying to display a very large text.

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Fixing problem with large text in the Custom Selection nodes.

### Reviewers

@QilongTang 

### FYIs

@avidit

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
This is a GIF showing the expected behavior for Custom Selection node.
RKRgSr8716

File: src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs [hunk 1]
```diff
@@ -27,12 +30,44 @@ public void CustomizeView(CustomSelection model, NodeView nodeView)
...
-            dropdown.Margin = new Thickness(40, 0, 0, 0);
+            dropdown.Margin = new Thickness(leftMargin, 0, 0, 0);
             dropdown.VerticalAlignment = VerticalAlignment.Top;
+            dropdown.ApplyTemplate();
+
+            var dropDownTextBlock = dropdown.Template.FindName("PART_ReadOnlyTextBlock", dropdown) as TextBlock;
+            if (dropDownTextBlock != null)
+            {
+                dropDownTextBlock.Visibility = Visibility.Visible;
+            }
```
[the remark this answers is not in the document]
Comment (RobertGlobant20) [reply src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs:47]: Updated to use a Property with the Converter.
See the commit: https://github.com/DynamoDS/Dynamo/pull/14960/commits/9721ab013e792b85d05b012b2523e46bfda36c47

File: src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs [hunk 2]
```diff
@@ -19,6 +21,7 @@ public class CustomSelectionNodeViewCustomization : DropDownNodeViewCustomizatio
         /// <param name="nodeView"></param>
         public void CustomizeView(CustomSelection model, NodeView nodeView)
         {
+            const double leftMargin = 40;
```
Comment (QilongTang) [inline src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs:24]: hmm where does this 40 come from?

File: src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (QilongTang) [reply src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs:47]: Thanks, a bit better. can you add comments to what dominates the value of  `IsVisibleDropDownTextBlock`

File: src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs [hunk 3]
```diff
@@ -27,12 +30,44 @@ public void CustomizeView(CustomSelection model, NodeView nodeView)
             base.CustomizeView(model, nodeView);
 
             var dropdown = (ComboBox)nodeView.inputGrid.Children[1];
+            dropdown.MaxWidth = formControl.Width - leftMargin;
 
             formControl.BaseComboBox = dropdown;
+            formControl.BaseComboBox.SelectionChanged += BaseComboBox_SelectionChanged;
```
Comment (QilongTang) [inline src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs:36]: unsubscribe somewhere?

File: src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs (hunk 2 above)
Comment (RobertGlobant20) [reply src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs:24]: It was already hardcoded I just move it to a const.

image

https://github.com/DynamoDS/Dynamo/blob/bb72d492d475b406bff9c50e46bac0f4ce050de9/src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs#L34

File: src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (RobertGlobant20) [reply src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs:52]: I've added more comments for clarifying the use of this property, please let me know if is clear enough.
For your second question, this is being done in the `CustomizeView()` method and I can see that this method is only called once when creating a CustomSelection node.
image

https://github.com/DynamoDS/Dynamo/pull/14960/commits/c08ab0e2f61b5605c254fd987dc8d385272623ea

File: src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs (hunk 3 above)
Comment (RobertGlobant20) [reply src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs:39]: For unsubscribing the method I added the Dispose() method but not sure if this class is being disposed correctly or not.
It's supposed that if you derive from IDispose and implement the Dispose() method the right way to dispose the object should be by using the "`using(instance)`" or by calling  `object.Dispose() `directly.
commit: https://github.com/DynamoDS/Dynamo/pull/14960/commits/c08ab0e2f61b5605c254fd987dc8d385272623ea

Comment (RobertGlobant20) [reply src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/CustomSelection.cs:39]: @QilongTang you were right, when closing the workspace the Dispose method is being reached.
image

**Review by QilongTang — APPROVED** [review]

**Pull request merged**